nodejs/awsx/ec2/vpc.ts

```diff
@@ -205,10 +233,122 @@ export class Vpc extends pulumi.ComponentResource {
      * This can be done by calling [subnet.createRoute] and passing in the newly created NatGateway.
      */
     public addNatGateway(name: string, args: x.ec2.NatGatewayArgs, opts: pulumi.ComponentResourceOptions = {}) {
-        const natGateway = new x.ec2.NatGateway(name, this, args, opts);
+        const natGateway = new x.ec2.NatGateway(name, this.parent, args, opts);
         this.natGateways.push(natGateway);
         return natGateway;
     }
+}
+
+(<any>VpcData).doNotCapture = true;
+utils.Capture(VpcData.prototype).addInternetGateway.doNotCapture = true;
+utils.Capture(VpcData.prototype).addNatGateway.doNotCapture = true;
+utils.Capture(VpcData.prototype).partition.doNotCapture = true;
+
+export class Vpc extends pulumi.ComponentResource {
+    public readonly id: pulumi.Output<string>;
+
+    public readonly vpc: pulumi.Output<aws.ec2.Vpc>;
+
+    private readonly _underlyingData: Promise<VpcData>;
+
+    constructor(name: string, args: VpcArgs | ExistingVpcArgs | ExistingVpcIdArgs, opts?: pulumi.ComponentResourceOptions);
+    /** @internal */
+    constructor(name: string, args: DefaultVpcArgs, opts?: pulumi.ComponentResourceOptions);
+    constructor(name: string, args: VpcArgs | ExistingVpcArgs | ExistingVpcIdArgs | DefaultVpcArgs, opts: pulumi.ComponentResourceOptions = {}) {
+        super("awsx:x:ec2:Vpc", name, {}, opts);
+
+        if (isExistingVpcArgs(args)) {
+            this._underlyingData = this.initializeExistingVpcArgs(name, args, opts);
+        }
+        else if (isDefaultVpcArgs(args)) {
+            this._underlyingData = this.initializeDefaultVpcArgs(name, args, opts);
+        }
+        else if (isExistingVpcIdArgs(args)) {
+            this._underlyingData = this.initializeExistingVpcIdArgs(name, args, opts);
+        }
+        else {
+            this._underlyingData = this.initializeVpcArgs(name, args, opts);
+        }
+
+        this.id = pulumi.output(this._underlyingData.then(v => v.id));
```

Comment:
this preserves semantics, but may not be what we want in the future.  specifically, someone who ends up depending on the vpc's id will only wait on the raw Vpc being created.  This is different than depending on the awsx.Vpc itself wwhere you will depend on all children being created.

Open question if we should have the same semantics when we lift inner outputs to the outer component.